pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -88,15 +88,20 @@ func (w *watcher) Run(ctx context.Context) error {
 	defer os.RemoveAll(workingDir)
 	w.workingDir = workingDir
 
-	for _, repoCfg := range w.config.Repositories {
-		repo, err := w.cloneRepo(ctx, repoCfg)
+	repoCfgs := make(map[string]config.PipedRepository, len(w.config.Repositories))
+	for _, repo := range w.config.Repositories {
+		repoCfgs[repo.RepoID] = repo
+	}
+	for _, r := range w.config.EventWatcher.GitRepos {
+		cfg := repoCfgs[r.RepoID]
```

Comment:
The specified ID could be wrong so we should check its existence.

Reply:
True